driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotSpawnMongocryptdTest.java

```diff
@@ -0,0 +1,243 @@
+            MongoDatabase db = client.getDatabase(KEY_VAULT_NAMESPACE.getDatabaseName());
+            db.drop();
+            db.getCollection(KEY_VAULT_NAMESPACE.getCollectionName(), BsonDocument.class)
+                    .withWriteConcern(WriteConcern.MAJORITY)
+                    .insertOne(EXTERNAL_KEY);
+        }
+    }
+
+    private static Map<String, Object> commonExtraAutoEncryptionOptions(final InetSocketAddress mongocryptdSocketAddress) {
+        return singletonMap("mongocryptdSpawnArgs", asList(
+                // We pick a random available `mongocryptd` port and also include it in the PID file path
+                // to reduce the chances of different test runs interfering with each other. The interference
+                // may come from the fact that once spawned, `mongocryptd` stays up and running for some time,
+                // which may cause failures in other runs if they use the same `mongocryptd` port / PID file.
+                format("--pidfilepath=bypass-spawning-mongocryptd-%d.pid", mongocryptdSocketAddress.getPort()),
+                format("--port=%d", mongocryptdSocketAddress.getPort())));
+    }
+
+    private void assertMongocryptdNotSpawned() {
+        try (MongoClient mongocryptdClient = createMongoClient(MongoClientSettings.builder()
+                .applyToClusterSettings(builder -> builder
+                        .hosts(singletonList(new ServerAddress(mongocryptdSocketAddress)))
+                        .serverSelectionTimeout(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS))
+                .build())) {
+            assertThrows(MongoTimeoutException.class, () -> mongocryptdClient.getDatabase(NAMESPACE.getDatabaseName())
+                    .runCommand(Document.parse("{hello: 1}")),
+                    "If nothing is thrown, then we connected to mongocryptd, i.e., it was spawned");
+        }
+    }
+
+    private static BsonDocument externalSchema() {
+        try {
+            return getTestDocument(new File(Assertions.assertNotNull(AbstractClientSideEncryptionNotSpawnMongocryptdTest.class
+                    .getResource("/client-side-encryption-external/external-schema.json")).toURI()));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static BsonDocument externalKey() {
+        try {
+            return getTestDocument(new File(Assertions.assertNotNull(AbstractClientSideEncryptionNotSpawnMongocryptdTest.class
+                    .getResource("/client-side-encryption-external/external-key.json")).toURI()));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SafeVarargs
+    @SuppressWarnings("varargs")
+    private static <K, V> Map<K, V> merge(final Map<K, V> map, final Entry<K, V>... entries) {
+        HashMap<K, V> result = new HashMap<>(map);
+        result.putAll(Stream.of(entries).collect(Collectors.toMap(Entry::getKey, Entry::getValue)));
```

Comment:
nit: extra Map created in collect. Might be
`for (Map.Entry<K, V> entry : entries) {
    result.put(entry.getKey(), entry.getValue());
}`